### PR TITLE
Refactor zoom-frame microstate to use spacemacs microstate macro.

### PR DESCRIPTION
**Pull request by ralesi**
Also Modify zoom in and out to keep frame the same size.

### CODE REVIEW

**Comment by syl20bnr** [comment]
I tested it in 24.3 and it triggers an error `let: Wrong number of arguments: set-frame-size, 4` but the zoom still works. However the frame is still resized.

File: spacemacs/extensions.el [hunk 1]
```diff
@@ -130,55 +130,43 @@
...
-          (powerline-reset))
-        (spacemacs/zoom-frame-overlay-map)
-        (spacemacs/zoom-frame-micro-state-doc))
-      (evil-leader/set-key
-        "zf+"  'spacemacs/zoom-in-frame
-        "zf-"  'spacemacs/zoom-out-frame
-        "zf="  'spacemacs/reset-zoom))))
-
-(defun spacemacs/init-emacs-builtin-process-menu ()
-  (evilify process-menu-mode process-menu-mode-map))
```
Comment (syl20bnr) [inline spacemacs/extensions.el]: Should not be deleted.

File: spacemacs/extensions.el [hunk 2]
```diff
@@ -130,55 +130,43 @@
...
-        "Zoom the buffer in/out. If DIRECTION is positive or zero the frame text is enlarged,
-otherwise it is reduced."
-        (interactive)
-        (cond
-         ((eq direction 0) (zoom-frm-unzoom))
-         ((< direction 0) (zoom-frm-out))
-         ((> direction 0) (zoom-frm-in)))
-        (when (fboundp 'powerline-reset)
-          (setq-default powerline-height (spacemacs/compute-powerline-height))
-          (powerline-reset))
```
Comment (syl20bnr) [inline spacemacs/extensions.el]: This is important to scale the powerline separators.

**Comment by ralesi** [comment]
Think I addressed the issues.  Need to work further on 24.3 compatibility.  Also, fhp is currently off slightly, so frame seems to shrink after repeated use.

**Comment by syl20bnr** [comment]
Thank you ! :+1: 
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**